tests/test_rst_compiler.py

```diff
@@ -1,238 +1,237 @@
+    """Test Vimeo iframe tag generation"""
+
+    sample = ".. vimeo:: VID\n   :height: 400\n   :width: 600"
+    html = get_html_from_rst(tempdir, sample)
+    assert_html_contains(
+        html,
+        "iframe",
+        attributes={
+            "src": ("https://player.vimeo.com/" "video/VID"),
+            "height": "400",
+            "width": "600",
+        },
+    )
+
+
+@pytest.mark.parametrize(
+    "sample",
+    [
+        ".. code-block:: python\n\n   import antigravity",
+        ".. sourcecode:: python\n\n   import antigravity",
+    ],
+)
+def test_rendering_codeblock_alias(tempdir, sample):
+    """Test CodeBlock aliases"""
+    get_html_from_rst(tempdir, sample)
+
+
+def test_doc_doesnt_exist():
+    with pytest.raises(Exception):
+        assert_html_contains("anything", {})
+
+
+def test_doc(tempdir):
+    sample = "Sample for testing my :doc:`fake-post`"
+    html = get_html_from_rst(tempdir, sample)
+    assert_html_contains(
+        html, "a", text="Fake post", attributes={"href": "/posts/fake-post"}
+    )
+
+
+def test_doc_titled(tempdir):
+    sample = "Sample for testing my :doc:`titled post <fake-post>`"
+    html = get_html_from_rst(tempdir, sample)
+    assert_html_contains(
+        html, "a", text="titled post", attributes={"href": "/posts/fake-post"}
+    )
+
+
+@pytest.fixture(autouse=True, scope="module")
+def localeborg_base():
+    """A base config of LocaleBorg."""
+    LocaleBorg.reset()
+    assert not LocaleBorg.initialized
+    LocaleBorg.initialize({}, "en")
+    assert LocaleBorg.initialized
+    assert LocaleBorg().current_lang == "en"
+    try:
+        yield
+    finally:
+        LocaleBorg.reset()
+        assert not LocaleBorg.initialized
+
+
+@pytest.fixture
+def tempdir(tmpdir):
+    """
+    Helper fixture to support Python 3.4 & 3.5 who are unable to work
+    with pathlib objects.
+    """
+    return str(tmpdir)
 
-    def test_test(self):
-        self.basic_test()
-        self.assertHTMLContains("iframe", attributes={"src": "foo"},
-                                text="spam")
-        self.assertRaises(Exception, self.assertHTMLContains, "eggs", {})
 
+def get_html_from_rst(temp_dir, rst):
+    """Create html output from rst string"""
 
-class MathTestCase(ReSTExtensionTestCase):
-    sample = r':math:`e^{ix} = \cos x + i\sin x`'
+    infile = os.path.join(temp_dir, "inf")
+    outfile = os.path.join(temp_dir, "outf")
 
-    def test_math(self):
-        """ Test that math is outputting TeX code."""
-        self.basic_test()
-        self.assertHTMLContains("span", attributes={"class": "math"},
-                                text=r"\(e^{ix} = \cos x + i\sin x\)")
+    with io.open(infile, "w+", encoding="utf8") as post_file:
+        post_file.write(rst)
 
+    compiler = nikola.plugins.compile.rest.CompileRest()
+    compiler.set_site(FakeSite())
+    compiler.site.post_per_input_file[infile] = FakePost(outfile)
+    compiler.compile(infile, outfile)
```

Comment:
Couldn’t we just call `compiler.compile_string` and avoid dealing with temp files?

Reply:
We get one rather unimportant line less covered but have less setup required for tests.
I'd say this is a good tradeoff!